external/netutils/netlib_netmon.c

```diff
@@ -0,0 +1,173 @@
+	char get_name[32];
+
+	printf("\nSockfd\tProto\tType\tLocal\t\t\tRemote\t\t\tstate\tPID\n");
+
+	for (i = 0; i < NUM_SOCKETS; ++i) {
+	/* Protect socket array */
+		if (sock[i].conn) {
+			printf("%-8d",i);
+			/* Protocol */
+			if (sock[i].conn->type & NETCONN_TCP) {
+				printf("TCP\t");
+#ifdef CONFIG_NET_IPv6
+				if (sock[i].conn->type & NETCONN_TYPE_IPV6) {
+					printf("IPv6\t");
+				} else {
+					printf("IPv4\t");
+				}
+#else
+				printf("IPv4\t");
+#endif
+				printf("%-15s:%-8s", inet_ntoa(sock[i].conn->pcb.ip->local_ip),"*");
+				printf("%-15s:%-8d", inet_ntoa(sock[i].conn->pcb.ip->remote_ip), sock[i].conn->pcb.tcp->remote_port);
+			} else if (sock[i].conn->type & NETCONN_UDP) {
+				printf("UDP\t");
+#ifdef CONFIG_NET_IPv6
+				if (sock[i].conn->type & NETCONN_TYPE_IPV6) {
+					printf("IPv6\t");
+				} else {
+					printf("IPv4\t");
+				}
+#else
+				printf("IPv4\t");
+#endif
+				printf("%-15s:%-8d", inet_ntoa(sock[i].conn->pcb.ip->local_ip),sock[i].conn->pcb.udp->local_port);
+				printf("%-15s:%-8d", inet_ntoa(sock[i].conn->pcb.ip->remote_ip), sock[i].conn->pcb.udp->remote_port);
+			} else {
+				printf("RAW\t");
+#ifdef CONFIG_NET_IPv6
+				if (sock[i].conn->type & NETCONN_TYPE_IPV6) {
+					printf("IPv6\t");
+				} else {
+					printf("IPv4\t");
+				}
+#else
+				printf("IPv4\t");
+#endif
+				printf("%-15s:%-8s", inet_ntoa(sock[i].conn->pcb.ip->local_ip),"*");
+				printf("%-15s:%-8s", inet_ntoa(sock[i].conn->pcb.ip->remote_ip),"*");
+			}
+
+			switch (sock[i].conn->state) {
+				case NETCONN_WRITE:
+					printf("WRITE\t");
+					break;
+				case NETCONN_LISTEN:
+					printf("LISTEN\t");
+					break;
+				case NETCONN_CONNECT:
+					printf("CONNECT\t");
+					break;
+				case NETCONN_CLOSE:
+					printf("CLOSE\t");
+					break;
+				default:
+					printf("NONE\t");
+					break;
+			}
+			ret_chk = pthread_getname_np(sock[i].conn->pid, get_name);
+			if (ret_chk == OK) {
+				printf("%d:%s\n", sock[i].conn->pid, get_name);
+			} else {
+				printf("%d:NONE\n", sock[i].conn->pid);
+			}
+		} 
+	}
+	return ;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: netlib_netmon
+ *
+ * Description:
+ *   Get the network driver IPv4 address
+ *
+ * Parameters:
+ *   command   Type of information to get
+ *
+ * Return:
+ *   0 on success; -1 on failure
+ *
+ ****************************************************************************/
+
+int netlib_netmon(FAR const char *command)
+{
+	int ret = ERROR;
+	FAR struct socket sockets_tmp[NUM_SOCKETS];
+	if (command) {
+		/* Get a socket (only so that we get access to the INET subsystem) */
+
+		int sockfd = socket(PF_INETX, NETLIB_SOCK_IOCTL, 0);
+		if (sockfd >= 0) {
+			if (strcmp(command, "all")) {
+				ret = ioctl(sockfd, SIOCGETSOCK, sockets_tmp);
```

Comment:
below 2 cases are same, please check it.

Reply:
Network monitor now provides getting only socket information. Other options (such as wifi stats; the number of wifi disconnect event) will be included.